src/devnet/external-server.ts

```diff
@@ -44,18 +45,20 @@ export async function getFreePort(): Promise<string> {
     );
 }
 
-export abstract class ExternalServer {
+export abstract class ExternalServer extends IntegratedDevnetLogger {
     protected childProcess: ChildProcess;
-    private lastError: string = null;
     private connected = false;
 
     constructor(
         protected host: string,
         protected port: string | null,
         private isAliveURL: string,
-        private processName: string
+        private processName: string,
+        protected stdout?: string,
+        protected stderr?: string
     ) {
-        ExternalServer.cleanupFns.push(this.cleanup.bind(this));
+            super(stdout, stderr);
```

Comment:
What happened to indentation here?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/179/commits/cfd12c44d12d31196d3a51293bdf70b855faa9fe